R/load.config.R

```diff
@@ -1,74 +1,148 @@
+library("pryr")
+library("utils")
+
 #' Loads a yaml config file
 #' @param project A project (use default config file names for this project)
 #' @param sp Subproject to activate
 #' @param file file path to config file, allows you to specify an exact file.
 #' @export
-load.config = function(project=NULL, sp=NULL, filename=NULL) {
-	if ( ! requireNamespace("yaml", quietly=TRUE)) {
-		warning("Package yaml is required to load yaml config files.")
-		return
-	}
+load.config = function(project=NULL, sp = NULL, filename=NULL, uses_paths_section = FALSE) {
+
+	# Derive project folder from environment variables and project name.
 	if (is.null(project)) { 
-		projectDir = options("PROJECT.DIR")
+		project_dir = options("PROJECT.DIR")
 	} else {
-		projectDir = paste0(Sys.getenv("CODEBASE"), project)
-	}
-	# If no file is specified, try these default locations
-	yamls = list("metadata/config.yaml",
-						"metadata/project_config.yaml",
-						paste0("metadata/", project, ".yaml"))
-	cfg = NULL
-	if (! is.null(filename)) {
-		yamls = c(filename, yamls)
+		codepath = Sys.getenv("CODE")
+		project_dir = if (is.null(codepath) | identical("", codepath)) project else file.path(codepath, project)
```

Comment:
Probably R is going to take a bit more work than python for standardization, since there is no PEP and nobody can agree on style. too much baggage, I think. The one I try to follow is bioconductor's: https://www.bioconductor.org/developers/how-to/coding-style/  with the exception that I can't stand `<-` instead of `=`. Maybe we need a general style tutorial in labDocs.